src/classes/LVL_LevelAssign_BATCH.cls

```diff
@@ -270,32 +287,106 @@ public without sharing class LVL_LevelAssign_BATCH implements Database.Batchable
     * @description query the Level object, and return a map of Object Name to a Set of Fields to query.
     * @return map<String, list<String>> The map of Object Name to its Set of Fields
     */
-    public static map<String, set<String>> getMapStrObjectToQueryFields() {
-        
-        // get all our levels
-        list<Level__c> listAllLvls = [select Id, Name, Target__c, Source_Field__c, Level_Field__c, 
-            Previous_Level_Field__c, Active__c
-            from Level__c
-            where Active__c = true
-            order by Target__c, Level_Field__c];
-            
-        // create our map
-        map<String, set<String>> mapStrObjectToQueryFields = new map<String, set<String>>();
+    public static List<Level> getLevels() {
+        List<Level__c> levels = [
+            SELECT Id, Target__c, Source_Field__c, Level_Field__c,
+                Previous_Level_Field__c, Active__c, LastModifiedDate
+            FROM Level__c
+            WHERE Active__c = true
+            ORDER BY Target__c, Level_Field__c];
+
+        List<Level> levelWrappers = new List<Level>();
+
+        for (Level__c aLevel : levels) {
+            levelWrappers.add(new Level(aLevel));
+        }
         
-        for (Level__c lvl : listAllLvls) {
-            // for each object, we'll add all its source, level, and previous level fields
-            String strObj = lvl.Target__c;
-            Set<String> setFlds = mapStrObjectToQueryFields.get(strObj);
-            if (setFlds == null) {
-                setFlds = new Set<String>();
-                mapStrObjectToQueryFields.put(strObj, setFlds);
+        return levelWrappers; 
+    }
+
+    /*********************************************************************************************************
+    * @description Wrapper class for the Level__c object
+    */
+    public class Level {
+        public String target { get { return levelRecord.Target__c; } }
+        public String sourceField { get { return levelRecord.Source_Field__c; } }
+        private Level__c levelRecord;
+        private Boolean targetIsAccount;
+        private Boolean targetIsContact;
+
+        /*********************************************************************************************************
+        * @description Constructor
+        */
+        public Level(Level__c levelRecord) {
+            this.levelRecord = levelRecord;
+            targetIsAccount = target == String.valueOf(Account.SObjectType);
```

Comment:
Rather than set the var in the constructor, could the var be defined as 
```
public Boolean targetIsAccount { 
   get { 
      return target == String.valueOf(Account.SObjectType);
   }
   private set; 
} 
```